magefile.go

```diff
@@ -3324,6 +3329,14 @@ func getOtelDependencies() (*otelDependencies, error) {
 		sort.Slice(list, func(i, j int) bool { return list[i].Name < list[j].Name })
 	}
 
+	// take care of replaces
+	for _, rep := range modFile.Replace {
```

Comment:
This looks correct, but since there are no replaces in the go.mod on main there's no proof it works. This feels like it shouldn't be too horrible to unit test, with a reference go.mod file. Are you able to add a simple test for at least the replace case?

Probably this entire set of functions should move into a sub-package of dev-tools to enable doing this.

Reply:
Thank you! LGTM